---
type: FilterCircuit
fields:
  start: ./1. Has Request Exceeded Throttling Limit?
  name: Throttle
  description: ""
children:
- type: MaximumNumberMsgsFilter
  fields:
    numberMessages: 3
    timePeriod: 15
    name: 1. Has Request Exceeded Throttling Limit?
  routing:
    success: ../2B. Connect to StockQuote Web Service
    failure: ../2A. Create Response Indicating Throttle Limit Exceeded
  logging:
    fatal: 'Throttle filter aborted: ${circuit.exception}'
    failure: Maximum number of messages filter was triggered
    success: Maximum number of messages filter was not triggered
- type: ConnectToURLFilter
  fields:
    url: http://stockquote.com/stockquote/instance1
    caCerts: []
    useRootCerts: false
    ssloptions:
    - nosslv2
    - nosslv3
    - notlsv1
    - notlsv1_1
    name: 2B. Connect to StockQuote Web Service
  routing:
    success: ../3B. Set HTTP Response Code to 200
  logging:
    fatal: 'Error occurred while routing request to endpoint. Error: ${circuit.exception}'
    failure: Failed to route request to endpoint.
    success: Successfully routed request to endpoint
  children:
  - type: SaveToFileFilter
    fields:
      name: Persistence Details
- type: Reflector
  fields:
    name: 3B. Set HTTP Response Code to 200
  logging:
    fatal: 'Error occurred while echoing the message. Error: ${circuit.exception}'
    failure: Failed to echo back the message
    success: Successfully echoed back the message
- type: ChangeMessageFilter
  fields:
    body: Warning!!  Throttle Limit has been exceeded
    outputContentType: text/plain
    name: 2A. Create Response Indicating Throttle Limit Exceeded
  routing:
    success: ../3A. Set HTTP Response Code to 500
  logging:
    fatal: 'Error in setting the message. Error: ${circuit.exception}'
    failure: Failed in setting the message
    success: Success in setting the message
- type: Reflector
  fields:
    httpResponseStatus: 500
    name: 3A. Set HTTP Response Code to 500
  logging:
    fatal: 'Error occurred while echoing the message. Error: ${circuit.exception}'
    failure: Failed to echo back the message
    success: Successfully echoed back the message
